### PR TITLE
can upvote and downvote on the webpage

### PR DESCRIPTION
**Pull request by SaskiaWells**

### CODE REVIEW

File: Utils/patchArticles.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import axios from 'axios'
+
+const NCnewsAPI = axios.create({
+  baseURL: "https://nc-news-tm65.onrender.com",
+});
+
+export const patchArticle = (id) => {
+return NCnewsAPI.patch(`/api/articles/${id}`)
```
Comment (cjplatten) [inline Utils/patchArticles.js]: There is no object being passed to the patch request here so this is not making a sucessful request to the api.

**Review by cjplatten — CHANGES_REQUESTED** [review]
While optiimistic rendering is workign fine, the changed vote counts are not persistent on refresh. See my comment on the api request.

File: Utils/patchArticles.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+});
+
+export const patchArticle = (id, obj) => {
+return NCnewsAPI.patch(`/api/articles/${id}`, obj)
+      .then((res) => {
+      return res.data;
+    })
+    .catch((err) => {
+      console.log(err);
+    });
```
Comment (aloush-dev) [inline Utils/patchArticles.js]: Shouldn't really have a catch here in your api file. If you need a catch chain it when you invoke the function in your components.

File: src/Article/votes.jsx [hunk 3]
```diff
@@ -1,6 +1,33 @@
...
+import { FaThumbsUp, FaThumbsDown } from "react-icons/fa";
+import { patchArticle } from "../../Utils/patchArticles";
+
+function Votes({ setArticle, article }) {
+
+
+    function handleUpClick() {
+        setArticle((currArticle) => {
+          return {...currArticle, votes:currArticle.votes + 1}
+        })
```
Comment (aloush-dev) [inline src/Article/votes.jsx]: Could we extract the votes out of the article into state so we don't have to update the whole article state?

**Review by aloush-dev — CHANGES_REQUESTED** [review]
Should I be able to upvote and downvote unlimited times?

If I change the URL to go to a different review or refresh the page it never loads and I am stuck on loading.

Check your console as there is a log showing, there is also a warning/error about a missing key.

Votes are also not staying after a refresh your api call must not be working properly.

**Review by cjplatten — CHANGES_REQUESTED** [review]
I would like to see some feedback on the button that they can no longer be clicked to add a vote once theey have been clicked. There is also an issue that when the down vote button is clicked, the number is optimistically rendered to go down but when the page is refreshed after this the number of votes is +1 on the previous. Have another look at your nadler function.

**Pull request merged**